### ISSUE TITLE
Why After hook is now shown on report ?

### ISSUE THREAD

**Issue by andreroggeri** [wswebcreation/multiple-cucumber-html-reporter#84]
Currently the reporter ignores hooks that fails. This cause some confusion to the users, because the CI is broken, but the report is green.

Can we add a option to consider these steps ?

**Comment by wswebcreation** [comment]
Hi @andreroggeri 

Can you provide an example of the JSON that should let the report fail? Normally if an after hooks fails it should also let the report fail and put it in the JSON as a failed status. So it could be that the status is not set by the framework that you are using.

Take for example this [file](https://github.com/wswebcreation/multiple-cucumber-html-reporter/blob/master/test/unit/data/json/failure_report_v3.json)

There you see the status of the After hook. If the status is set to passed and the rest of the steps are also set to passed the module will set the report status to passed

```json
          {
            "keyword": "After",
            "hidden": true,
            "match": {
              "location": "e2e-tests/config/helpers/after.scenario.ts:23"
            },
            "result": {
              "status": "passed",
              "duration": 667
            },
            "embeddings": [
              {
                "data": "",
                "media": {
                  "type": "image/png"
                }
              }
            ]
          }
```

**Comment by andreroggeri** [comment]
There you go.

From what I was reading on the code, it seems to be necessary to debug the logic on the [following line](https://github.com/wswebcreation/multiple-cucumber-html-reporter/blob/78489fb26f0dadfec34566d369702e1ea942d60d/lib/generate-report.js#L359)

```json
[
  {
    "keyword": "Feature",
    "name": "Test",
    "line": 1,
    "id": "test",
    "tags": [],
    "uri": "src\\test.feature",
    "elements": [
      {
        "id": "test;login",
        "keyword": "Scenario",
        "line": 3,
        "name": "Login",
        "tags": [
          {
            "name": "@Test",
            "line": 2
          }
        ],
        "type": "scenario",
        "steps": [
          {
            "arguments": [],
            "keyword": "Given ",
            "line": 4,
            "name": "i am on agendaodonto login page",
            "match": {
              "location": "src\\demo.steps.ts:26"
            },
            "result": {
              "status": "passed",
              "duration": 2836000000
            }
          },
          {
            "arguments": [],
            "keyword": "When ",
            "line": 5,
            "name": "I enter with valid credentials",
            "match": {
              "location": "src\\demo.steps.ts:30"
            },
            "result": {
              "status": "passed",
              "duration": 575000000
            }
          },
          {
            "arguments": [],
            "keyword": "Then ",
            "line": 6,
            "name": "I see the homepage",
            "match": {
              "location": "src\\demo.steps.ts:34"
            },
            "result": {
              "status": "passed",
              "duration": 1000000
            }
          },
          {
            "keyword": "After",
            "hidden": true,
            "match": {
              "location": "src\\demo.steps.ts:22"
            },
            "result": {
              "status": "failed",
              "duration": 1000000,
              "error_message": "Error: This is a error created by miself\n    at World.<anonymous> (E:\\Projetos\\protractor-test\\src\\demo.steps.ts:23:11)"
            }
          },
          {
            "keyword": "After",
            "hidden": true,
            "match": {
              "location": "node_modules\\protractor-cucumber-framework\\lib\\resultsCapturer.js:26"
            },
            "result": {
              "status": "passed",
              "duration": 1000000
            }
          }
        ]
      }
    ]
  }
]
```

**Comment by wswebcreation** [comment]
Thanks for this, I need to debug this. There is a reason for this logic, just need to find it again 😉 . I also need to check what the `hidden` property means. Will get back on it

**Comment by wswebcreation** [comment]
I released version `1.11.8` https://github.com/wswebcreation/multiple-cucumber-html-reporter/releases/tag/v1.11.8 to fix this problem.

Going to close the issue now
